opal/mca/common/ofi/common_ofi.c

```diff
@@ -533,8 +540,9 @@ get_nearest_nics(int *num_distances, pmix_value_t **valin)
     PMIx_Info_destruct(&directive);
     if (ret != PMIX_SUCCESS || !val) {
         ret = compute_dev_distances(&distances, &ndist);
```

Comment:
Does it segfault every time without `--enable-debug`? Or only sometimes?

Reply:
> Does it segfault every time without --enable-debug? Or only sometimes?

Every time. I have a sense that we should probably address this issue separately. This PR in itself solves immediate issues that are blocking our internal development. I see value to merge it first.

Reply:
are you running it as a singleton? or through MPI? I see a crash when running it as a singleton.

Reply:
I believe this PR was supposed to fix the singleton problem. Are you saying you still see it? Or that you see it when debug is disabled, but only when run as a singleton?

Reply:
@rhc54, I was asking @wenduwan. I have not verified if this PR fixes the singleton case.

Reply:
@rhc54 although I'm not clear how this fixes the singleton crash. It seems like it would occur in the PMIX_Get_cpuset(). At least that's what I was seeing before.

Reply:
@amirshehataornl I don't think it's singleton - I added this print
```
[1,37]<stderr>: [queue-c6inmetal-st-c6inmetal-1:38605] common_ofi.c:752:opal_process_info.is_singleton: 0
```

Reply:
> It seems like it would occur in the PMIX_Get_cpuset(). At least that's what I was seeing before.

Yes - and this PR (the combination of the changes to opal PLUS the PMIx and PRRTE updates) fixes that problem.

Reply:
I believe the problem was that the `pmix_topo` variable was not initialized prior to passing it to the PMIx library. Thus, when debug was enabled, it happened to have a `NULL` in the source string - but had garbage when debug was disabled, causing PMIx to crash.

Reply:
@rhc54 Thanks! I can confirm the latest update fixed the segfault.